org.osgi.test.assertj.cm/src/main/java/org/osgi/test/assertj/cm/configurationevent/AbstractConfigurationEventAssert.java

```diff
@@ -0,0 +1,235 @@
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's pid is not
+	 *             null.
+	 */
+	public S hasNoPid() {
+
+		isNotNull().has(ConfigurationEventConditions.pidNull());
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's pid is equal to the given
+	 * one.
+	 *
+	 * @param pid the given pid to compare the actual ConfigurationEvent's pid
+	 *            to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's pid is not
+	 *             equal to the given one.
+	 */
+	public S hasPid(String pid) {
+
+		isNotNull().has(ConfigurationEventConditions.pid(pid));
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's reference is equal to the
+	 * given one.
+	 *
+	 * @param reference the given reference to compare the actual
+	 *            ConfigurationEvent's reference to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's reference is
+	 *             not equal to the given one.
+	 */
+	public S hasReference(org.osgi.framework.ServiceReference<ConfigurationAdmin> reference) {
+		// check that actual ConfigurationEvent we want to make assertions on is
+		// not null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting reference of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";
+
+		// null safe check
+		org.osgi.framework.ServiceReference<ConfigurationAdmin> actualReference = actual.getReference();
+		if (!Objects.deepEquals(actualReference, reference)) {
+			failWithMessage(assertjErrorMessage, actual, reference, actualReference);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	public ServiceReferenceAssert hasReferenceThat() {
+		return ServiceReferenceAssert.assertThat(actual.getReference())
+			.as(actual + ".reference");
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's type matches to the given
+	 * type.
+	 *
+	 * @param type the given type to compare the actual ConfigurationEvent's
+	 *            type to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's type is not
+	 *             matching to the given one.
+	 */
+	public S hasType(int type) {
+
+		isNotNull().has(ConfigurationEventConditions.type(type));
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual ConfigurationEvent's type is
+	 * <code>ConfigurationEvent.CM_DELETED</code>.
+	 *
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual ConfigurationEvent's type is not
+	 *             equal to <code>ConfigurationEvent.CM_DELETED</code>.
+	 */
+	public S hasTypeDeleted() {
```

Comment:
I am not sure having a method per type make sense when you supply the type to `hasTypeEqualTo`.

Reply:
removed